backend/app/quiz/schemas.py

```
@@ -0,0 +1,105 @@
+
+class QuizAnswerSubmit(BaseModel):
+    """Schema for submitting quiz answer."""
+    user_id: int = Field(..., description="User ID")
+    question_id: int = Field(..., description="Question ID")
+    chosen_option: str = Field(..., description="Chosen option (A, B, C, D)")
+    response_time_sec: Optional[float] = Field(None, ge=0, description="Time taken in seconds")
+    
+    @validator('chosen_option')
+    def validate_option(cls, v):
+        if v.upper() not in ['A', 'B', 'C', 'D']:
+            raise ValueError('Option must be A, B, C, or D')
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)